src/Talis/Persona/Client/Login.php

```diff
@@ -61,9 +61,19 @@ public function requireAuth($provider, $appId, $appSecret, $redirectUri = '')
      */
     public function validateAuth()
     {
+        if (isset($_POST['persona:signature']))
+        {
+            $payloadSignature = $_POST['persona:signature'];
+        } else {
+            $this->getLogger()->error("Signature not set");
+            throw new \Exception('Signature not set');
```

Comment:
Ah, very good catch!

This code is not used by any of our apps, so depot is the only app affected, and I haven't been catching exceptions there, so what happens currently is that you get a 500 error.

Not sure if a 401 here is the right thing to do though? Because it is a problem with how we're trying to do login or might be a problem with Persona itself, not really "Unauthorized." Or do you think this should be "Unauthorised."

In any case, it's probably better to introduce this as a persona client exception (e.g. `LoginException`) and let the apps handle the errors how they choose to, no?